### PR TITLE
feat: Implemented RPC Manager for RPC calls

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Implemented an RPC Manager which handles RPC selection at the start selecting the best RPC out of the available endpoints. Proceeding further it handles the switching and refreshing of RPC endpoints in a sorted order of there performance.

Also introduces a `blockMonitor` in `block` module which is resposnsible for providing block number to the logs and also check for stale blocks from the current best RPC endpoint.
This is implemented in https://github.com/razor-network/oracle-node/pull/126.

Introduced a `importEndpoints` command which imports the list of endpoints provided in `core/endpoints.go` to file `$HOME/.rzr/endpoints`.
 
Fixes https://linear.app/interstellar-research/issue/RAZ-1026

# How Has This Been Tested?

Ran a staker with vote command on testnet by providing multiple active and inactive RPCs.
Analysed the refresh periods, switching when RPC is down, and when RPC endpoints are not imported.

### CODE REVIEW

**Comment by Yashk767** [comment]
Rename `RPC` module to `rpc`

File: README.md [hunk 1]
```diff
@@ -190,6 +190,23 @@ Password:
 
 _Before staking on Razor Network, please ensure your account has sFUEL and RAZOR. For testnet RAZOR, please contact us on Discord._
 
+### Import Endpoints
+
+You can import the endpoints mentioned in `endpoints.json` from the repository to your `$HOME/.razor/endpoints.json` file using the `importEndpoints` command.
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply README.md]: done in https://github.com/razor-network/oracle-node/pull/1260/commits/faa6865644108591d99145d2e0491aaf89ad77e6

File: RPC/rpc_struct.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	URL         string
+	BlockNumber uint64
+	Latency     float64
+	Client      *ethclient.Client
+}
+
+type RPCManager struct {
+	Endpoints     []*RPCEndpoint
+	mutex         sync.RWMutex
+	BestRPCClient *ethclient.Client // Holds the current best RPC client
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply RPC/rpc_struct.go]: done in https://github.com/razor-network/oracle-node/pull/1260/commits/faa6865644108591d99145d2e0491aaf89ad77e6

**Comment by Yashk767** [comment]
> 5631ae4

Done in https://github.com/razor-network/oracle-node/pull/1260/commits/5631ae4a3bf2be50403596b5afdf033bdf0a1e18 @ashish10677

File: README.md [hunk 3]
```diff
@@ -190,6 +190,23 @@ Password:
 
 _Before staking on Razor Network, please ensure your account has sFUEL and RAZOR. For testnet RAZOR, please contact us on Discord._
 
+### Import Endpoints
+
+You can import the endpoints to file `$HOME/.razor/endpoints.json` on your local by using the `importEndpoints` command.
+This multiple providers along with the user input provider are sorted according to the best performance and the best provider is chosen by the RPC manager which will be used to make the RPC calls.
```
Comment (ashish10677) [inline README.md:196]: I think this sentence is wrong. There should be something after `This`. This provides multiple providers or copies multiple providers.

Comment (Yashk767) [reply README.md:196]: okay, will correct it.

Comment (Yashk767) [reply README.md]: Done in https://github.com/razor-network/oracle-node/pull/1260/commits/42e8cd0e9235a988f36447629bf0ef89247c255f

**Review by ashish10677 — APPROVED** [review]

**Review by rajkharvar — APPROVED** [review]

**Pull request merged**